packages/coreutils/src/interfaces.ts

```diff
@@ -15,7 +15,7 @@ import { ISignal } from '@lumino/signaling';
 /**
  * A generic interface for change emitter payloads.
  */
-export interface IChangedArgs<T, U extends string = string> {
+export interface IChangedArgs<T, OldT = T, U extends string = string> {
```

Comment:
This commit adds an optional generic arg to allow change notifications to specify different typings for the old and new values (nice e.g. when the initial value is `null`, but it can never be set back to `null` after that). Note that this changes the order of the generic args (the name typing gets pushed later).